### PR TITLE
A few changes to reduce memory usage (T2696)

### PR DESCRIPTION
**Pull request by ajcollison**

### CODE REVIEW

**Comment by arincos** [comment]
Once the unit tests pass, LGTM!

Let me know if you need any help with the unit tests.

**Comment by ajcollison** [comment]
Unit tests fix.  Please take a look at those changes, though, to make sure I fixed them in a good way :)

**Comment by arincos** [comment]
Great fixes to the unit tests.

LGTM!

File: radiometric_normalization/pif.py [hunk 1]
```diff
@@ -246,12 +248,49 @@ def _PCA_fit_single_band(cand_valid, ref_valid):
     return pca
 
 
+def np_array_from_2arrays(array1, array2, dtype=numpy.uint16):
+    '''Efficiently combine two 1-D arrays into a single 2-D array.
```
Comment (jreiberkyle) [inline radiometric_normalization/pif.py:252]: excellent documentation!

**Pull request merged**